File: backend/src/main/java/com/project/dto/mood/response/MoodAnalyticsResponseDTO.java
package com.project.dto.mood.response;

import java.util.List;
import java.util.Map;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class MoodAnalyticsResponseDTO {

    private double averageMood;
    private int totalEntries;

    // For line / bar charts
    private List<MoodDailyStatsDTO> dailyStats;

    // For pie charts
    private Map<String, Long> moodDistribution;

}

File: backend_dotnet/SafeMind.Core/DTOs/Mood/Response/MoodAnalyticsResponseDTO.cs
using System;
using System.Collections.Generic;

namespace SafeMind.Core.DTOs.Mood.Response
{
    public class MoodAnalyticsResponseDTO
    {
        public double AverageMood { get; set; }
        public int TotalEntries { get; set; }

        // For line / bar charts
        public List<MoodDailyStatsDTO> DailyStats { get; set; } = new List<MoodDailyStatsDTO>();

        // For pie charts
        public Dictionary<string, long> MoodDistribution { get; set; } = new Dictionary<string, long>();
    }
}
